src/projects/shared/api/circuit/src/public/impl/Component.ts

```diff
@@ -115,6 +115,7 @@ export class ComponentImpl<T extends CircuitTypes> extends BaseObjectImpl<T> imp
     }
 
     public toSchema(): Schema.Component {
-        return ({ ...this.getComponent() });
+        const comp = this.getComponent();
+        return ({ ...comp, props: { ...comp.props } });
```

Comment:
I'm assuming this is because we don't want a shallow copy of props?

Reply:
yeah